--- a/modules/gui/qt/components/controller.cpp
+++ b/modules/gui/qt/components/controller.cpp
@@ -812,6 +812,17 @@ FullscreenControllerWidget::FullscreenCo
 
     vout.clear();
 
+#ifdef QT5_HAS_WAYLAND
+    if( b_hasWayland )
+    {
+        // Popup is less than perfect in that it seems impossible to make it non-modal
+        // and you can't get it to fade but at least it goes where it is asked to and
+        // does less confusing things with other popups
+        setWindowFlags( Qt::Popup | Qt::FramelessWindowHint);
+        setWindowModality( Qt::NonModal );
+    }
+    else
+#endif
     setWindowFlags( Qt::Tool | Qt::FramelessWindowHint | Qt::X11BypassWindowManagerHint );
     setAttribute( Qt::WA_ShowWithoutActivating );
     setMinimumWidth( FSC_WIDTH );
